src/odemis/util/test/util_test.py

```diff
@@ -396,54 +401,255 @@ def test_line_clipping(self):
+        in_exp = {
+            tuple([-1, -5, -6]): 0,
+            tuple([1, 2, 3]): 2,
+            tuple([0, 5, 10]): 1,
+            tuple([-15, 30, -60]): 0
+        }
+
+        for testlist, hardcoded_closest_index in in_exp.items():
+            closest = util.index_closest(5, testlist)
+            self.assertEqual(hardcoded_closest_index, closest,  "Failed to get correct output for "
+                                                                "the closest index to 5 in %s" % in_exp)
+
+
+class NormalizeRectTestCase(unittest.TestCase):
+
+    def test_already_normalized(self):
+        """ Test normalizing a rectangle with already normalized values """
+        rect = (1, 2, 3, 4)
+        normalized_rect = util.normalize_rect(rect)
+        self.assertEqual(normalized_rect, (1, 2, 3, 4), "Failed to normalize the rectangle")
+
+    def test_non_normalized(self):
+        """ Test normalizing a rectangle with not yet normalized values """
+        rect = (10, 7, 5, 3)
+        normalized_rect = util.normalize_rect(rect)
+        self.assertEqual(normalized_rect, (5, 3, 10, 7), "Failed to normalize the rectangle")
+
+
+class IsPointInRectTestCase(unittest.TestCase):
+
+    def test_points_in_rect(self):
+        """ Test whether the given coordinates are in the rectangle """
+        rect = (5, 3, 10, 7)
+        point = (8.0, 4.0)
+        in_rect = util.is_point_in_rect(point, rect)
+        self.assertTrue(in_rect, "Failed to get correct output for the point in rectangle %s with "
+                                 "point %s" % (rect, point))
+
+    def test_points_outside_rect(self):
+        """ Test whether the given coordinates are not in the rectangle """
+        rect = (5, 3, 10, 7)
+        point = (22.0, 44.0)
+        in_rect = util.is_point_in_rect(point, rect)
+        self.assertFalse(in_rect, "Failed to get correct output for the point in rectangle %s with "
+                                  "point %s" % (rect, point))
+
+
+class ExpandRectTestCase(unittest.TestCase):
+
+    def test_expand_rect(self):
+        """ Test expanding a rectangle """
+        rect = (5, 3, 10, 7)
+        expanded_rect = util.expand_rect(util.normalize_rect(rect), 5)
+        expected_rect = (0, -2, 15, 12)
+        self.assertEqual(expanded_rect, expected_rect,  "Failed to get correct output for rectangle %s with "
+                                                        "expanded rectangle %s" % (expanded_rect, expected_rect))
+
+
+class FindPlotContentTestCase(unittest.TestCase):
+
+    def test_find_plot_content(self):
+        """ Test finding left and right most non-zero values"""
+        xd = (10, 15, 20, 25, 30)
+        yd = (0, 0, 1, -2, 0)
+        plot_content = util.find_plot_content(xd, yd)
+        self.assertEqual(plot_content, (20, 25), "Failed to get correct output while finding plot content "
+                                                 "for %s with values  %s %s" % (plot_content, xd, yd))
```

Comment:
I think it would also make sense to add a test where there is one or more zero in between the left and right most non-zero value. So something like this for xd and yd:
```
        xd = (10, 15, 16, 17, 20, 25, 30)
        yd = (0, 0, 1, 0, 0, -2, 0)
```

Reply:
or should yd also be sorted?